internal/provider/machine_resource.go

```diff
@@ -168,8 +168,11 @@ func (mr flyMachineResourceType) GetSchema(context.Context) (tfsdk.Schema, diag.
 					},
 					"volume": {
 						Required:            true,
-						MarkdownDescription: "Name or ID of volume",
+						MarkdownDescription: "ID of volume",
```

Comment:
If name did work it technically still will, I just suggest not documenting it because the ID will work better - it will change if the `fly_volume` is replaced, so we'll detect the change here; if it's recreated with the same `name` then that wouldn't have changed and wouldn't force a replacement here, so better to use ID.